### PR TITLE
dev(python): Add support for python 3

### PR DESCRIPTION
**Pull request by NiklasMM**

### CODE REVIEW

**Comment by NiklasMM** [comment]
@klugjohannes @MTraub80 Would you please take a look? I tried to get this package ready for Python3

**Comment by NiklasMM** [comment]
Yes! And please stick to English as this is a public repo :stuck_out_tongue:

File: tests/test_parse.py [hunk 1]
```diff
@@ -2,40 +2,42 @@
...
-        str(message)
-        unicode(message)
+import six
+import sys
+import pytest
+
+@pytest.mark.skipif(sys.version_info.major > 2, reason="not relevant for python 3")
+def test_bytestring_segment():
+    """ bytestring segment can be cast to unicode and bytestring """
+    segment = HL7Segment(b"FOO|Baababamm")
```
Comment (klugjohannes) [inline tests/test_parse.py]: Why did you remove the Umlauts?

**Comment by klugjohannes** [comment]
@NiklasMM you should also update the README.md accordingly.

File: tests/test_parse.py (hunk 1 above)
Comment (NiklasMM) [reply tests/test_parse.py]: It was in an effort to make these tests run in Py3, before I decided they were not relevant. 
Still, having non-ASCII characters in a `b` string doesn't really make sense, does it?

Comment (klugjohannes) [reply tests/test_parse.py]: Yupp, true for the latter. And they are really not relevant in py3, so we're safe there. In Python2 this should probably be a string-Type and `b'string'` was used to enforce the type in a unicode_literals environment. Test docstring was probably misleading. Here's the original commit: ec632dd1924a5ded35f2155e665d7cc8cd5a27a4

File: hl7parser/hl7_data_types.py [hunk 2]
```diff
@@ -134,9 +139,9 @@ def __getitem__(self, idx):
         return self.list_[idx]
 
     def __unicode__(self):
-        return self.delimiters.rep_separator.join(map(unicode, self.list_))
+        return self.delimiters.rep_separator.join(map(six.text_type, self.list_))
 
-    def __str__(self):
+    def __str__(self): # pragma: no cover
         return self.__unicode__()
```
Comment (MTraub80) [inline hl7parser/hl7_data_types.py]: ..hier auch wieder der Hinweis wie oben

File: hl7parser/hl7_data_types.py [hunk 3]
```diff
@@ -115,6 +118,8 @@ def __nonzero__(self):
         elif len(self.input_fields) > 1:
             return True
 
+    def __bool__(self): # pragma: no cover
+        return self.__nonzero__()
```
Comment (MTraub80) [inline hl7parser/hl7_data_types.py]: Das verstehe ich nicht. Warum brauchen wir das jetzt?

**Comment by NiklasMM** [comment]
@klugjohannes I updated the Readme
@MTraub80 I included a UnicodeMixin class, thanks for the tip!

File: hl7parser/hl7_data_types.py (hunk 3 above)
Comment (NiklasMM) [reply hl7parser/hl7_data_types.py]: Die Magic method für `bool(x)` war in Python 2 `__nonzero__` und ist jetzt `__bool__`
